app/src/main/java/org/p2p/wallet/striga/signup/ui/StrigaSignUpFirstStepPresenter.kt

```diff
@@ -91,53 +134,35 @@ class StrigaSignUpFirstStepPresenter(
         countryOfBirth = interactor.findCountryByIsoAlpha3(
             data[StrigaSignupDataType.COUNTRY_OF_BIRTH]?.value.orEmpty()
         )
+        selectedCountryCode = countryCodeRepository.findCountryCodeByPhoneCode(
+            data[StrigaSignupDataType.PHONE_CODE]?.value.orEmpty()
+        )
         countryOfBirth?.let { onCountryChanged(it) }
     }
 
     private fun mapDataForStorage() {
-        val fullPhoneNumber = getFullPhoneNumber()
-        if (fullPhoneNumber != null && phoneMask != null) {
-            val phoneCodeWithSign = phoneMask?.phoneCodeWithSign ?: throw IllegalStateException("Phone mask is null")
-            setData(
-                StrigaSignupDataType.PHONE_CODE,
-                phoneCodeWithSign
-            )
-            setData(
-                StrigaSignupDataType.PHONE_NUMBER,
-                removeCodeFromPhoneNumber(phoneCodeWithSign, fullPhoneNumber)
-            )
-        }
         setData(StrigaSignupDataType.COUNTRY_OF_BIRTH, countryOfBirth?.codeAlpha3.orEmpty())
     }
 
-    private suspend fun setupPhoneMask() {
-        val country = Country(
-            name = "Turkey",
-            flagEmoji = "",
-            codeAlpha2 = "TR",
-            codeAlpha3 = "TUR"
-        ) // interactor.getSelectedCountry()
-
-        phoneMask = interactor.findPhoneMaskByCountry(country)
-        // todo: check that all countries have phone mask
-        phoneMask?.let {
-            val maskForFormatter = "+" + it.mask
-            view?.setPhoneMask(maskForFormatter)
-        }
-    }
+    private suspend fun loadDefaultCountryCode() {
+        try {
+            val countryCode: CountryCode? =
+                countryCodeRepository.detectCountryCodeBySimCard()
+                    ?: countryCodeRepository.detectCountryCodeByNetwork()
+                    ?: countryCodeRepository.detectCountryCodeByLocale()
```

Comment:
We already have the same logic inside the `CountryRepository#detectCountryOrDefault()` (`CountryInMemoryRepository`). Make a copy of this function inside the CountryCodeLocalRepository and then reuse it in the `CountryRepository#detectCountryOrDefault()` to return `Country`

Reply:
I don't know why you created second repository

Reply:
It wasn't me 😄